src/App.js

```diff
@@ -14,11 +17,113 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const preparedGoods = goodsFromServer.map((good, index) => (
+  {
+    name: good,
+    id: index,
+  }
+));
+
+class App extends React.Component {
+  state = {
+    isShown: false,
+    goods: preparedGoods,
+    selected: 1,
+  };
+
+  clickHandler = () => {
+    this.setState({
+      isShown: true,
+    });
+  }
```

Comment:
можно сделать универсальным и менять на противоположное `!`
код компонента может расти и уже непонятно что за clickHandler, к какой кнопке относится
```suggestion
  handleStart = () => {
    this.setState((state) => {
      started: !state.started,
    });
  }
```